packages/player/src/components/player.tsx

```diff
@@ -0,0 +1,461 @@
+                isActive: (_context, _event) => true,
+              }),
+            },
+            VOLUME_CHANGE: {
+              target: 'playing',
+              actions: [
+                assign({
+                  volume: (_context, event) => event.volume,
+                }),
+                (context, event) => {
+                  if (context.video) context.video.volume = event.volume
+                },
+              ],
+            },
+            PLAYBACKRATE_CHANGE: {
+              target: 'playing',
+              actions: [
+                assign({
+                  playbackRate: (_context, event) => event.playbackRate,
+                }),
+                (context, event) => {
+                  if (context.video)
+                    context.video.playbackRate = event.playbackRate
+                },
+              ],
+            },
+            SEEKING: {
+              target: 'playing',
+              actions: [
+                assign({
+                  seekingTime: (context, event) => event.seekingTime,
+                }),
+                'seekVideo',
+              ],
+            },
+            END: 'ended',
+            TIMING: {
+              target: 'playing',
+              actions: assign({
+                currentTime: (context, _event) =>
+                  context.video?.currentTime ?? 0,
+                duration: (context, _event) => context.video?.duration ?? 0,
+              }),
+            },
+          },
+        },
+        ended: {on: {PLAY: 'playing'}},
+      },
+    },
+    failure: {},
+  },
+})
+
+interface VideoContextType {
+  videoService: ActorRefFrom<typeof videoMachine>
+}
+
+export const VideoContext = createContext({} as VideoContextType)
+
+export const VideoProvider: React.FC = (props) => {
+  const videoService = useInterpret(videoMachine, {
+    actions: {
+      playVideo: (context, _event) => {
+        const {video} = context
+        video?.play()
+      },
+      pauseVideo: (context, _event) => {
+        const {video} = context
+        video?.pause()
+      },
+      seekVideo: (context, _event) => {
+        const {video, seekingTime} = context
+        if (video) video.currentTime = seekingTime ?? video.currentTime
+      },
+    },
+  })
+
+  return (
+    <VideoContext.Provider value={{videoService}}>
+      {props.children}
+    </VideoContext.Provider>
+  )
+}
+
+export const Player: React.FC = ({children}) => {
+  const {videoService} = React.useContext(VideoContext)
+  return (
+    <div
+      onMouseDown={() => videoService.send('ACTIVITY')}
+      onMouseMove={() => videoService.send('ACTIVITY')}
+      onKeyDown={() => videoService.send('ACTIVITY')}
+      className="cueplayer-react"
+      style={{height: '500px'}}
+    >
+      <div className="cueplayer-react-controls-enabled">
+        <Video>{children}</Video>
+      </div>
+      <VideoControlBar />
+      <Shortcut />
+    </div>
+  )
+}
+
+const VideoControlBar = () => {
```

Comment:
just inlining the primary components for now and not worried too much about component composition at this phase